lib/ui/compositing/scene_host.cc

```diff
@@ -48,30 +48,37 @@ flutter::SceneHost* GetSceneHost(scenic::ResourceId id) {
   return nullptr;
 }
 
-void InvokeDartClosure(tonic::DartPersistentValue* closure) {
-  if (closure) {
-    std::shared_ptr<tonic::DartState> dart_state = closure->dart_state().lock();
-    if (!dart_state) {
-      return;
-    }
-
-    tonic::DartState::Scope scope(dart_state);
-    tonic::DartInvoke(closure->value(), {});
+void InvokeDartClosure(const tonic::DartPersistentValue& closure) {
+  auto dart_handle = closure.value();
+  FML_DCHECK(Dart_IsClosure(dart_handle));
+  if (Dart_IsNull(dart_handle)) {
```

Comment:
Can this also be a DCHECK? And then guard the call-sites below against passing null instead.